include/sound/sof/xman.h

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __INCLUDE_SOUND_SOF_XMAN_H__
+#define __INCLUDE_SOUND_SOF_XMAN_H__
+
+/* In ASCII XMan (byte order) */
+#define XMAN_MAGIC_NUMBER	0x6e614d58
+
+/* Build u32 number in format MMmmmppp */
+#define XMAN_BUILD_VERSION(MAJOR, MINOR, PATH) ((u32)( \
+	(UL(MAJOR) << 24) | \
+	(UL(MINOR) << 12) | \
+	UL(PATH)))
+
+/* check extended manifest version consistency */
+#define XMAN_VERSION_INCOMPATIBLE(host_ver, cli_ver) ( \
+	((host_ver) & GENMASK(31, 24)) != \
+	((cli_ver) & GENMASK(31, 24)))
+
+/* used extended manifest header version */
+#define XMAN_VERSION		XMAN_BUILD_VERSION(1, 0, 0)
+
+/* extended manifest header, deleting any field breaks backward compatibility */
+struct xman_header {
+	uint32_t magic;		///< identification number, XMAN_MAGIC_NUMBER
+	uint32_t size;		///< sum of header and content size
+	uint32_t header_size;	///< makes header extensionable
+	uint32_t version;	///< header version, not related with content
+	uint8_t content[0];	///< list of ext_man_elem_* elements
+};
+
+/* extended manifest element header */
+struct xman_elem_header {
+	uint32_t size;		///< in bytes, including header size
+	uint32_t type;		///< EXT_MAN_
+	uint8_t data[0];	///< code dependent content
+};
+
+/* Now define extended manifest elements */
+
+/* Extended manifest elements identificators */
+enum xman_elem_type {
+}
```

Comment:
empty enum ??

Reply:
rather not

Reply:
not

Reply:
To fully introduce extended manifest mechanism separately from introducing extended manifest elements in following commits